system/session/drivers/cookie.php

```diff
@@ -47,7 +48,7 @@ public function write($id, $cargo) {
 		$serialized = serialize($cargo);
 
 		// create a signature to verify content when unpacking
-		$sign = hash('md5', $serialized);
+		$sign = hash_hmac('sha256', $serialized, $session_key, true);
```

Comment:
> if the stored password is in md5, wouldn't comparing an md5 hash and an sha256 hash return false always?

Yes, but this isn't really for passwords. The md5 hash is currently taken over the old session data on write then retrieved on read. If it fails, reset the session.

This won't create a feedback loop because the next time they set the cookie, it will use SHA256.